pkg/security/ebpf/c/bind.h

```diff
@@ -0,0 +1,109 @@
+#ifndef _BIND_H_
+#define _BIND_H_
+
+struct bind_event_t {
+    struct kevent_t event;
+    struct process_context_t process;
+    struct span_context_t span;
+    struct container_context_t container;
+    struct syscall_t syscall;
+
+    int socket;
+    uint16_t addr_family;
+    uint16_t addr_port;
+    union  {
+        uint32_t addr_ip;
+        uint8_t  addr_ip6[16];
+    };
+};
+
+SYSCALL_KPROBE3(bind, int, socket, struct sockaddr*, addr, unsigned int, addr_len) {
+    if (!addr) {
+        return 0;
+    }
+
+    struct policy_t policy = fetch_policy(EVENT_BIND);
+    if (is_discarded_by_process(policy.mode, EVENT_BIND)) {
+        return 0;
+    }
+
+    /* cache the bind and wait to grab the retval to send it */
+    struct syscall_cache_t syscall = {
+        .type = EVENT_BIND,
+        .bind = {
+            .socket = socket,
+            .addr = addr,
+            .addr_len = addr_len,
+        },
+    };
+    cache_syscall(&syscall);
+    return 0;
+}
+
+SYSCALL_KRETPROBE(bind) {
+    struct syscall_cache_t *syscall = pop_syscall(EVENT_BIND);
+    if (!syscall) {
+        return 0;
+    }
+
+    int retval = PT_REGS_RC(ctx);
+    if (IS_UNHANDLED_ERROR(retval)) {
+        return 0;
+    }
+
+
+    /* get address family */
+    uint16_t addr_family;
+    if (bpf_probe_read(&addr_family, sizeof(addr_family), &(syscall->bind.addr->sa_family))) {
+        return 0;
+    }
+
+    /* pre-fill the event */
+    struct bind_event_t event = {
+        .syscall.retval = retval,
+        .socket = syscall->bind.socket,
+        .addr_family = addr_family,
+        .addr_port = 0,
+    };
+
+    /* get additionnal information, depending on the addr family */
+    struct sockaddr_in* sockin;
+    struct sockaddr_in6* sockin6;
+    switch (addr_family) {
```

Comment:
This entire section seems to be a duplicate of `kprobe/security_socket_bind` in `flow.h`. You might want to keep only one of them / refactor the kprobe hook point if you want to make the code cleaner :)
This also means that you probably don't need to put anything in the syscall_cache, but I would still keep the syscall hook points so that we know that this bind event comes from a syscall.